src/Data/Nat/Binary/Subtraction.agda

```diff
@@ -0,0 +1,344 @@
+1+[2 x ] ∸ 1+[2 y ] = double (x ∸ y)
+2[1+ x ] ∸ 1+[2 y ] with x <? y
+... | yes _ = 0ᵇ
+... | no _  = 1+[2 (x ∸ y) ]
+1+[2 x ] ∸ 2[1+ y ] with x ≤? y
+... | yes _ = 0ᵇ
+... | no _  = pred (double (x ∸ y))
+
+------------------------------------------------------------------------
+-- Properties of _∸_ and _≤_/_<_
+
+x∸0≡x : ∀ x → x ∸ 0ᵇ ≡ x
+x∸0≡x zero     = refl
+x∸0≡x 2[1+ _ ] = refl
+x∸0≡x 1+[2 _ ] = refl
+
+even∸odd-for≥ : ∀ {x y} → x ≥ y → 2[1+ x ] ∸ 1+[2 y ] ≡ 1+[2 (x ∸ y) ]
+even∸odd-for≥ {x} {y} x≥y with x <? y
+... | no _    = refl
+... | yes x<y = contradiction x≥y (<⇒≱ x<y)
+
+odd∸even-for> : ∀ {x y} → x > y → 1+[2 x ] ∸ 2[1+ y ] ≡ pred (double (x ∸ y))
+odd∸even-for> {x} {y} x>y with x ≤? y
+... | no _    = refl
+... | yes x≤y = contradiction x>y (≤⇒≯ x≤y)
+
+x≤y⇒x∸y≡0 : ∀ {x y} → x ≤ y → x ∸ y ≡ 0ᵇ
+x≤y⇒x∸y≡0 {zero} {_} _           = refl
+x≤y⇒x∸y≡0 (inj₁ (even<even x<y)) = x≡0⇒double[x]≡0 (x≤y⇒x∸y≡0 (inj₁ x<y))
+x≤y⇒x∸y≡0 {2[1+ x ]} {2[1+ x ]} (inj₂ refl) =
+  x≡0⇒double[x]≡0 (x≤y⇒x∸y≡0 {x} {x} (inj₂ refl))
+x≤y⇒x∸y≡0 {2[1+ x ]} {1+[2 y ]} (inj₁ (even<odd x<y)) with x <? y
+... | yes _  = refl
+... | no x≮y = contradiction x<y x≮y
+x≤y⇒x∸y≡0 {1+[2 x ]} {2[1+ y ]} (inj₁ (odd<even x≤y)) with x ≤? y
+... | yes _  = refl
+... | no x≰y = contradiction x≤y x≰y
+x≤y⇒x∸y≡0 {1+[2 _ ]} {1+[2 _ ]} (inj₁ (odd<odd x<y)) =
+  x≡0⇒double[x]≡0 (x≤y⇒x∸y≡0 (inj₁ x<y))
+x≤y⇒x∸y≡0 {1+[2 x ]} {1+[2 x ]} (inj₂ refl) =
+  x≡0⇒double[x]≡0 (x≤y⇒x∸y≡0 (≤-refl {x}))
+
+x∸x≡0 : ∀ x → x ∸ x ≡ 0ᵇ
+x∸x≡0 x = x≤y⇒x∸y≡0 (≤-refl {x})
+
+x>y⇒x∸y≢0 : ∀ {x y} → x > y → x ∸ y ≢ 0ᵇ
+x>y⇒x∸y≢0 {x'@(2[1+ x ])} {zero} _ = subst (_≢ 0ᵇ) (sym (x∸0≡x x')) 2[1+x]≢0
+x>y⇒x∸y≢0 {2[1+ _ ]} {2[1+ _ ]} (even<even y<x) = x≢0⇒double[x]≢0 (x>y⇒x∸y≢0 y<x)
+x>y⇒x∸y≢0 {2[1+ x ]} {1+[2 y ]} (odd<even y≤x) with x <? y
+... | no _    = 1+[2x]≢0 {x ∸ y}
+... | yes x<y = contradiction y≤x (<⇒≱ x<y)
+x>y⇒x∸y≢0 {x'@(1+[2 x ])} {y'@(2[1+ y ])} (even<odd y<x) =
+  subst (_≢ 0ᵇ) (pred[2d]≡x'∸y') pred[2d]≢0
+  where
+  -- d = x ∸ y
+  d≢0            = x>y⇒x∸y≢0 y<x
+  pred[2d]≡x'∸y' = sym (odd∸even-for> y<x)
+  1≤d            = x<y⇒suc[x]≤y (x≢0⇒x>0 d≢0)
+  2≤2d           = double-mono-≤ 1≤d
+  1≤pred[2d]     = pred-mono-≤ 2≤2d
+  0<pred[2d]     = suc[x]≤y⇒x<y 1≤pred[2d]
+  pred[2d]≢0     = >⇒≢ 0<pred[2d]
+x>y⇒x∸y≢0 {1+[2 _ ]} {1+[2 _ ]} (odd<odd y<x) = x≢0⇒double[x]≢0 (x>y⇒x∸y≢0 y<x)
+
+x∸y≡0⇒x≤y : ∀ {x y} → x ∸ y ≡ 0ᵇ → x ≤ y
+x∸y≡0⇒x≤y {x} {y} x∸y≡0 with x ≤? y
+... | yes leq = leq
+... | no x≰y  = contradiction x∸y≡0 (x>y⇒x∸y≢0 (≰⇒> x≰y))
+
+x<y⇒0<y∸x : ∀ {x y} → x < y → 0ᵇ < y ∸ x
+x<y⇒0<y∸x {x} {y} x<y with <-cmp (y ∸ x) 0ᵇ
+... | tri> _ _     gt = gt
+... | tri≈ _ y∸x≡0 _  = contradiction (x∸y≡0⇒x≤y y∸x≡0) (<⇒≱ x<y)
+... | tri< y∸x<0 _ _  = contradiction y∸x<0 x≮0
+
+---------------------------------------------------------------
+-- Properties of _∸_ and _+_ and _≤_
+
+x≥y⇒[x∸y]+y≡x : ∀ {x y} → x ≥ y → (x ∸ y) + y ≡ x
+x≥y⇒[x∸y]+y≡x {zero} {y} y≤0 = begin-equality
```

Comment:
Pattern match on `y≤0` and this whole thing becomes `refl`

Reply:
This sentence is replaced with                                                              
```                                                                                         
x≥y⇒[x∸y]+y≡x {zero} {_} (inj₂ refl) = refl                                                 
```